src/loaders/lottie/tvgLottieModel.h

```diff
@@ -431,6 +432,7 @@ struct LottieGradient : LottieObject
 
         //color remains
         while (cidx < clast) {
+            if (cidx + 3 >= color.input->count) break;
```

Comment:
Yes, we can test the `test_IDX_984_RAND_1864499262634970735.json`.
Without this condition, buffer-overflow occurred in different line.

```cpp
cs.b = lroundf((*color.input)[cidx + 3] * 255.0f);
```

<img width="950" alt="Screenshot 2024-04-03 at 10 35 23 AM" src="https://github.com/thorvg/thorvg/assets/11167117/6c1eeb8c-cbd1-471e-ae3b-3c48cd8133ea">

Reply:
So, 'still +3' indicates the value of the blue channel, I assume. In this way, I guess this could also be +2 or +1, stemming from invalid input data. I believe this prevent should be addressed fundamental level.

Reply:
@hermet I see.
In this case, the `ColorStop& color` has only 3 values.
<img width="682" alt="Screenshot 2024-04-03 at 11 07 55 AM" src="https://github.com/thorvg/thorvg/assets/11167117/5009613c-b14a-4fe5-8fce-8a50d6ab8162">

Should we exit the populate function if given `color->input.count` can't be divided by number 4? becuase the `color remaining` requires 4 values(_offset, r, g, b_) from input every loop.

Like `if (!color.input || color.input->count % 4 != 0) return 0;` at line 386.

Reply:
@tinyjin I agree.  probably, we can log there like:
if (color.input->count % 4 != 0) TVGERR("LOTTIE", "the given colorstop value is not correct!");